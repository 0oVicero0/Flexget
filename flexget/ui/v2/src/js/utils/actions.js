export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
