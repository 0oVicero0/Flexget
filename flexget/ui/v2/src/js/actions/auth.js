import * as utils from 'utils/actions';
import { get, post } from 'utils/fetch';

export const AUTH = 'AUTH';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

const createAction = utils.createAction(AUTH);
const loading = utils.createLoading(AUTH);

export function checkLogin() {
  return (dispatch) => {
    get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
  };
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    get('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .then(err => dispatch(createAction(LOGOUT, err)));
  };
}
