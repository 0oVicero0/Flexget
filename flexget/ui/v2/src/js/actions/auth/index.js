import { createAction, loading } from 'utils/actions';
import { get, post } from 'utils/fetch';

const PREFIX = '@flexget/auth/';
export const LOGIN = `${PREFIX}LOGIN`;
export const LOGOUT = `${PREFIX}LOGOUT`;

export function checkLogin() {
  return dispatch => get('/server/version')
      .then(() => dispatch(createAction(LOGIN)))
      .catch(() => {});
}

export function login(credentials) {
  return (dispatch) => {
    dispatch(loading(LOGIN));
    return post('/auth/login', credentials)
      .then(() => dispatch(createAction(LOGIN)))
      .catch(err => dispatch(createAction(LOGIN, err)));
  };
}

export function logout() {
  return (dispatch) => {
    dispatch(loading(LOGOUT));
    return post('/auth/logout')
      .then(() => dispatch(createAction(LOGOUT)))
      .catch(err => dispatch(createAction(LOGOUT, err)));
  };
}
