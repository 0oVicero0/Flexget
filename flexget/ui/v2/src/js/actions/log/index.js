import oboe from 'oboe';
import { createAction } from 'utils/actions';

const PREFIX = '@flexget/log/';
export const LOG_START = `${PREFIX}LOG_START`;
export const LOG_MESSAGE = `${PREFIX}LOG_MESSAGE`;
export const LOG_DISCONNECT = `${PREFIX}LOG_DISCONNECT`;
export const LOG_LINES = `${PREFIX}LOG_LINES`;
export const LOG_QUERY = `${PREFIX}LOG_QUERY`;
export const LOG_CLEAR = `${PREFIX}LOG_CLEAR`;

export function startLogStream() {
  return (dispatch, getState) => {
    const { lines, query } = getState().log;
    const stream = oboe({
      url: `api/server/log?lines=${lines}&search=${query}`,
      method: 'GET',
    });

    const promise = new Promise((resolve, reject) => {
      stream
        .on('start', () => dispatch(createAction(LOG_START)))
        .on('node', '{message}', message => dispatch(createAction(LOG_MESSAGE, message)))
        .done(() => resolve())
        .fail(err => reject(err));
    });

    promise.abort = () => {
      stream.abort();
      dispatch(createAction(LOG_DISCONNECT));
    };

    return promise;
  };
}

export function setLines(lines) {
  return createAction(LOG_LINES, lines);
}

export function setQuery(query) {
  return createAction(LOG_QUERY, query);
}

export function clearLogs() {
  return createAction(LOG_CLEAR);
}

