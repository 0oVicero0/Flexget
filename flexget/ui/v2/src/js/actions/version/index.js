import * as utils from 'utils/actions';
import { get } from 'utils/fetch';

export const VERSION = 'VERSION';
export const GET_VERSION = 'GET_VERSION';

const createAction = utils.createAction(VERSION);
const loading = utils.createLoading(VERSION);

export function getVersion() {
  return (dispatch) => {
    dispatch(loading(GET_VERSION));
    return get('/server/version')
      .then(version => dispatch(createAction(GET_VERSION, version)))
      .catch(err => dispatch(createAction(GET_VERSION, err)));
  };
}
