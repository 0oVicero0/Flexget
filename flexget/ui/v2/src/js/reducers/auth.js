import { LOGIN, LOGOUT } from 'actions/auth';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    default:
      if (action.error && action.payload && action.payload.status === 401) {
        return {};
      }

      return state;
  }
}
