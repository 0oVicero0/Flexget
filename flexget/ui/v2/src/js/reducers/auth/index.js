import { LOGIN, LOGOUT } from 'actions/auth';
import { ERROR_STATUS } from 'actions/status';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    case ERROR_STATUS:
      if (action.payload.statusCode === 401) {
        return {};
      }
    default: // eslint-disable-line no-fallthrough
      return state;
  }
}
