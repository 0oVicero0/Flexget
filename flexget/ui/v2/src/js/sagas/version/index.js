import { call, put, takeLatest } from 'redux-saga/effects';
import { action, requesting } from 'utils/actions';
import { get } from 'utils/fetch';
import { GET_VERSION } from 'actions/version';

export function* getVersion() {
  try {
    const { data } = yield call(get, '/server/version');
    yield put(action(GET_VERSION, data));
  } catch (err) {
    yield put(action(GET_VERSION, err));
  }
}

export default function* saga() {
  yield takeLatest(requesting(GET_VERSION), getVersion);
}
