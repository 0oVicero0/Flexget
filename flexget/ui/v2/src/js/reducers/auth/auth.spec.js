import reducer from 'reducers/auth';
import { LOGIN, LOGOUT } from 'actions/auth';
import { ERROR_STATUS } from 'actions/status';
import { GET_VERSION } from 'actions/version';

describe('reducers/auth', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should login on LOGIN', () => {
    expect(reducer(undefined, { type: LOGIN })).toMatchSnapshot();
  });

  it('should login on GET_VERSION', () => {
    expect(reducer(undefined, { type: GET_VERSION })).toMatchSnapshot();
  });

  it('should logout on LOGOUT', () => {
    expect(reducer({ loggedIn: true }, { type: LOGOUT })).toMatchSnapshot();
  });

  it('should logout on a 401 ERROR_STATUS', () => {
    expect(reducer({ loggedIn: true }, {
      type: ERROR_STATUS,
      payload: {
        statusCode: 401,
      },
    })).toMatchSnapshot();
  });

  it('should stay logged In on a non 401 ERROR_STATUS', () => {
    expect(reducer({ loggedIn: true }, {
      type: ERROR_STATUS,
      payload: {
        statusCode: 404,
      },
    })).toMatchSnapshot();
  });
});
