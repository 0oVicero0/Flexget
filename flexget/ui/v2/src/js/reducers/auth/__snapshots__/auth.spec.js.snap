// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/auth should login on LOGIN 1`] = `
Object {
  "loggedIn": true,
}
`;

exports[`reducers/auth should logout on LOGOUT 1`] = `Object {}`;

exports[`reducers/auth should logout on a 401 ERROR_STATUS 1`] = `Object {}`;

exports[`reducers/auth should return the initial state 1`] = `Object {}`;

exports[`reducers/auth should stay logged In on a non 401 ERROR_STATUS 1`] = `
Object {
  "loggedIn": true,
}
`;
